cmd/uzfs_test/uzfs_txg_diff.c

```diff
@@ -0,0 +1,320 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+#include <sys/zfs_context.h>
+#include <sys/txg.h>
+#include <sys/zil.h>
+#include <sys/uzfs_zvol.h>
+#include <uzfs_mgmt.h>
+#include <uzfs_mtree.h>
+#include <uzfs_io.h>
+#include <uzfs_test.h>
+
+extern int total_time_in_sec;
+extern void populate_data(char *buf, uint64_t offset, int idx,
+    uint64_t block_size);
+extern uint64_t block_size;
+
+static int del_from_mblktree(avl_tree_t *tree, uint64_t b_offset,
+    uint64_t b_len);
+static int uzfs_search_mblktree(avl_tree_t *tree, uint64_t offset,
+    uint64_t *len);
+
+typedef struct wblkinfo {
+	uint64_t offset;
+	uint64_t len;
+	list_node_t link;
+} wblkinfo_t;
+
+int
+del_from_mblktree(avl_tree_t *tree, uint64_t offset, uint64_t len)
```

Comment:
operations on tree can go into one file in lib/libzpool/

Reply:
in production code, we are not using this API. So, i didn't keep it in lib/libzpool. Once we need this, we will move this to lib/libzpool.